### DISCUSSION TITLE
General suggestions about styling "custom" stuff

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [JakeStanger/ironbar#31] [General]
# Sys-Info
https://github.com/JakeStanger/ironbar/wiki/Sys-Info
Currently, it's only possible to rice the whole part of sys-info, not individual sections of it.
For example, I have these three modules.
![image](https://user-images.githubusercontent.com/68589851/198599025-b5c23cf3-006d-4a64-a9b0-ca1e30473e99.png)
I can't style them individualy, only by the full class `#sysinfo #item`

# Scripts
You can only have one style.(according to the documentation, it's very much possible that I read this incorrectly)
https://github.com/JakeStanger/ironbar/wiki/script

Maybe a variable like `class` in custom would help?
https://github.com/JakeStanger/ironbar/wiki/custom

**Comment by JakeStanger** [comment]
CSS provides an `nth-child` selector which you can use to target individual elements. I don't think having a `class` property would be preferable here because it would complicate the config - you'd end up having to specify an object for each segment instead of just a string like you do now.

Example:

```css
#sysinfo #item:nth-child(1) {
  /* this is your cpu */
}

#sysinfo #item:nth-child(2) {
  /* this is your memory */
}
```
https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child

You can even do some basic formulae with it, which I believe GTK supports.

**Reply by JustSimplyKyle** [reply]
Oh, very interesting! I'm not too good at css, thanks!

**Reply by JustSimplyKyle** [reply]
Slight issue? I can't seem to get it to work with multiple script?

**Reply by JakeStanger** [reply]
Please share your CSS otherwise I can't help

**Reply by JustSimplyKyle** [reply]
https://0x0.st/HrbE.css

**Reply by JakeStanger** [reply]
Okay so:

`nth-child` is not the nth of that type, but the nth of *any* type in the parent. That means `script#nth-child(1)` will only select script modules that are the first module in the section (start, center, end). 

Regular web CSS has a `nth-of-type` selector that would solve this, but GTK helpfully doesn't support it. For now, you can just bump up the numbers to their actual indexes in the section, and I'll make a note to add some options to make this easier.
